src/generators/impulsegenerator.cpp

```diff
@@ -0,0 +1,21 @@
+#include "impulsegenerator.h"
+
+using namespace eDSP::generators;
+
+template<typename T>
+const std::vector<T> &ImpulseGenerator<T>::generate() {
+
+    Generator<T>::data[0] = Generator<T>::amplitude;
+
+    for (size_t i = 1, size = Generator<T>::data.size(); i < size; i++) {
```

Comment:
There is no need to do this operation in every callback, its better to set all the buffer to zeros, (by default is like that) and update only the given amplitude in every single execution. As we can modify the size and the amplitude during runtime, then I have added some modifications to set all buffers to zero by default while resizing the vector. See this commit for details: https://github.com/mohabouje/eDSP/commit/abc2e441c5424da8efeb20f0680b91739efa8d82

Reply:
What if I don't want to change either the size, or the amplitude? Am I sure the data will be initialized to zero?
`Math::Container::set(data, static_cast<T>(0));` is called only when the size changes...?

Reply:
Is called in the default constructor with the different parameters too, so there are to options to create the generator, by setting the size in the constructor or by setting it calling setSize, both se the data array to zero in all values. In any case the idea is to have the algorithtemplate that check if an algorithm is initialised correctly, so if you don't call the correct constructor, the state is false. The idea is to have a clean interface that check if an algorithm has been initialised correctly and give you the error description, see the AlgorithTemplate class for details.

Reply:
In any case, my idea is to add a macro that removes the copy, move and default empty constructor in this kind of algorithms, then we will force the user to set up the algorithm
 allways with a given size.

Reply:
Done:  https://github.com/mohabouje/eDSP/commit/c0fb6079ca460772f193088f06f32facbcf9f656

Now a generator should be set up with a size at least.

Reply:
OK, I understand the macros EDSP_DISABLE_COPY, EDSP_DISABLE_DEFAULT now.